### PR TITLE
[TypeScript Client] fix install Aurelia + fix use deprecated function

### PR DESCRIPTION
**Pull request by karismann**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.
@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10)

### Description of the PR

- fix install Aurelia (upgrade and add missing dependencies)
- fix use deprecated function  in typescript client codegen

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/typescript-aurelia/package.json.mustache [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-    "test": "tslint api.ts"
+    "build": "tsc --outDir dist/",
+    "prepare": "npm run build",
+    "test": "tslint Api.ts"
   },
   "devDependencies": {
     "tslint": "^3.15.1",
-    "typescript": "^1.8.10",
-    "typings": "^1.0.4"
+    "typescript": "^2.4"
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-aurelia/package.json.mustache]: is typescript 3.x also allowed?

Comment (karismann) [reply modules/openapi-generator/src/main/resources/typescript-aurelia/package.json.mustache]: it seem to work with latest typescript version so i'll update the PR with it

File: modules/openapi-generator/src/main/resources/typescript-aurelia/package.json.mustache [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+  "scripts": {
+    "build": "tsc --outDir dist/",
+    "prepare": "npm run build",
+    "test": "tslint Api.ts"
   },
   "devDependencies": {
     "tslint": "^3.15.1",
-    "typescript": "^1.8.10",
-    "typings": "^1.0.4"
+    "typescript": "^3.3.1"
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/resources/typescript-aurelia/package.json.mustache]: please allow older typescript versions, too,
```suggestion
    "typescript": "^2.4 || ^3.0"
```

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java [hunk 3]
```diff
@@ -512,7 +513,7 @@ public String toApiName(String name) {
         if (name.length() == 0) {
             return "DefaultService";
         }
-        return initialCaps(name) + serviceSuffix;
+        return StringUtils.camelize(name) + serviceSuffix;
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java]: I'll file a PR to deprecate initialCaps before the 4.x release.

**Pull request merged**